cmake/macros/Public.cmake

```diff
@@ -1043,21 +1043,20 @@ function(pxr_toplevel_epilogue)
         # that we carefully avoid adding the usd_m target itself by using
         # TARGET_FILE.  Linking the usd_m target would link usd_m and
         # everything it links to.
-        
-        if(MSVC)
+        if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
             target_link_libraries(usd_ms
                 PRIVATE
-                    -WHOLEARCHIVE:$<BUILD_INTERFACE:$<TARGET_FILE:usd_m>>
+                    -Wl,-force_load $<BUILD_INTERFACE:$<TARGET_FILE:usd_m>>
             )
-        elseif(CMAKE_COMPILER_IS_GNUCXX)
+        elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU" AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
```

Comment:
Portability: Not clear why `Linux` only specific here albeit the documented snippet has this check. It would be poignant to ensure this is tested with GCC , Clang, or Clang-Cl (CMake >= 3.15) under Windows.

Reply:
Further documentation on `WHOLE_ARCHIVE` validates this platform specificity. https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_LINK_LIBRARY_USING_FEATURE.html#predefined-features

The MSVC  `clang-cl` toolchain is the only chain not handled by this legacy logic, however CMake 3.15 is natively supoirted by USD at this time so can be deemed out-of-scope under review.
👍